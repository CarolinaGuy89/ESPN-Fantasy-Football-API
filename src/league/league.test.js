import ApiModel from '../api-model/api-model.js';
import League from './league.js';

import { localObject, serverResponse } from './league.stubs.js';

describe('League', () => {
  let league;

  beforeEach(() => {
    league = new League();
  });

  afterEach(() => {
    league = null;
  });

  test('extends ApiModel', () => {
    expect(league).toBeInstanceOf(ApiModel);
  });

  describe('attribute population from server response', () => {
    beforeEach(() => {
      league = League.buildFromServer(serverResponse);
    });

    test('parses data correctly', () => {
      expect(league).toMatchSnapshot();
    });
  });

  describe('attribute population from local object', () => {
    beforeEach(() => {
      league = League.buildFromLocal(localObject);
    });

    test('parses data correctly', () => {
      expect(league).toMatchSnapshot();
    });
  });

  describe('instance methods', () => {
    describe('when params are passed', () => {
      test('adds seasonId to params passed to super read', () => {
        const seasonId = 2017;
        league.seasonId = seasonId;

        // Super lazy way to test
        jest.spyOn(ApiModel.prototype, 'read').mockImplementation();

        league.read({ params: { some: 'params' } });

        expect(ApiModel.prototype.read).toBeCalledWith({
          params: {
            seasonId,
            some: 'params'
          },
          model: league,
          route: League.route,
          reload: true
        });

        ApiModel.prototype.read.mockRestore();
      });
    });

    describe('when not params are passed', () => {
      test('passes seasonId as params to super read', () => {
        const seasonId = 2017;
        league.seasonId = seasonId;

        // Super lazy way to test
        jest.spyOn(ApiModel.prototype, 'read').mockImplementation();

        league.read();

        expect(ApiModel.prototype.read).toBeCalledWith({
          params: { seasonId },
          model: league,
          route: League.route,
          reload: true
        });

        ApiModel.prototype.read.mockRestore();
      });
    });
  });
});
