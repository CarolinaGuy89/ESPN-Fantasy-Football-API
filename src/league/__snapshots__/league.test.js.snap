// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`League attribute population from local object parses data correctly 1`] = `
League {
  "allowsTrades": true,
  "divisions": Array [
    Object {
      "divisionId": 0,
      "name": "Division 1",
      "size": 4,
    },
    Object {
      "divisionId": 1,
      "name": "Division 2",
      "size": 4,
    },
  ],
  "draftOrder": Array [
    Team {
      "abbreviation": "GOAT",
      "divisionStanding": 1,
      "firstName": "The Boston",
      "lastName": "TE Party",
      "leagueId": 336358,
      "leagueStanding": 1,
      "losses": 4,
      "pointsAgainst": 1131.4,
      "pointsFor": 1305.78,
      "seasonId": 2017,
      "teamId": 1,
      "ties": 0,
      "waiverRank": 6,
      "wins": 9,
    },
    Team {
      "abbreviation": "GRON",
      "divisionStanding": 4,
      "firstName": "Prestige",
      "lastName": "Worldwide",
      "leagueId": 336358,
      "leagueStanding": 8,
      "losses": 8,
      "pointsAgainst": 1232.94,
      "pointsFor": 1062.82,
      "seasonId": 2017,
      "teamId": 2,
      "ties": 0,
      "waiverRank": 5,
      "wins": 5,
    },
    Team {
      "abbreviation": "ARCH",
      "divisionStanding": 4,
      "firstName": "Archie",
      "lastName": "Cooper",
      "leagueId": 336358,
      "leagueStanding": 7,
      "losses": 8,
      "pointsAgainst": 1280,
      "pointsFor": 1155.34,
      "seasonId": 2017,
      "teamId": 3,
      "ties": 0,
      "waiverRank": 8,
      "wins": 5,
    },
    Team {
      "abbreviation": "REES",
      "divisionStanding": 3,
      "firstName": "Feed",
      "lastName": "Zeeeke ",
      "leagueId": 336358,
      "leagueStanding": 4,
      "losses": 6,
      "pointsAgainst": 1290.12,
      "pointsFor": 1297.26,
      "seasonId": 2017,
      "teamId": 4,
      "ties": 0,
      "waiverRank": 7,
      "wins": 7,
    },
    Team {
      "abbreviation": "fuck",
      "divisionStanding": 2,
      "firstName": "fuck fantasy ",
      "lastName": "football",
      "leagueId": 336358,
      "leagueStanding": 5,
      "losses": 7,
      "pointsAgainst": 1278.42,
      "pointsFor": 1233.26,
      "seasonId": 2017,
      "teamId": 5,
      "ties": 0,
      "waiverRank": 4,
      "wins": 6,
    },
    Team {
      "abbreviation": "BOIS",
      "divisionStanding": 3,
      "firstName": "Team",
      "lastName": "ASS",
      "leagueId": 336358,
      "leagueStanding": 6,
      "losses": 8,
      "pointsAgainst": 1240.04,
      "pointsFor": 1328.1,
      "seasonId": 2017,
      "teamId": 8,
      "ties": 0,
      "waiverRank": 1,
      "wins": 5,
    },
    Team {
      "abbreviation": "��",
      "divisionStanding": 2,
      "firstName": "2 Gurleys",
      "lastName": "One Cup",
      "leagueId": 336358,
      "leagueStanding": 3,
      "losses": 5,
      "pointsAgainst": 1186.94,
      "pointsFor": 1288.64,
      "seasonId": 2017,
      "teamId": 9,
      "ties": 0,
      "waiverRank": 2,
      "wins": 8,
    },
    Team {
      "abbreviation": "YOC",
      "divisionStanding": 1,
      "firstName": "FRICKIN",
      "lastName": "JOOOOOSH!",
      "leagueId": 336358,
      "leagueStanding": 2,
      "losses": 6,
      "pointsAgainst": 1250.94,
      "pointsFor": 1219.6,
      "seasonId": 2017,
      "teamId": 12,
      "ties": 0,
      "waiverRank": 3,
      "wins": 7,
    },
  ],
  "finalRankings": Array [
    Team {
      "abbreviation": "GOAT",
      "divisionStanding": 1,
      "firstName": "The Boston",
      "lastName": "TE Party",
      "leagueId": 336358,
      "leagueStanding": 1,
      "losses": 4,
      "pointsAgainst": 1131.4,
      "pointsFor": 1305.78,
      "seasonId": 2017,
      "teamId": 1,
      "ties": 0,
      "waiverRank": 6,
      "wins": 9,
    },
    Team {
      "abbreviation": "GRON",
      "divisionStanding": 4,
      "firstName": "Prestige",
      "lastName": "Worldwide",
      "leagueId": 336358,
      "leagueStanding": 8,
      "losses": 8,
      "pointsAgainst": 1232.94,
      "pointsFor": 1062.82,
      "seasonId": 2017,
      "teamId": 2,
      "ties": 0,
      "waiverRank": 5,
      "wins": 5,
    },
    Team {
      "abbreviation": "ARCH",
      "divisionStanding": 4,
      "firstName": "Archie",
      "lastName": "Cooper",
      "leagueId": 336358,
      "leagueStanding": 7,
      "losses": 8,
      "pointsAgainst": 1280,
      "pointsFor": 1155.34,
      "seasonId": 2017,
      "teamId": 3,
      "ties": 0,
      "waiverRank": 8,
      "wins": 5,
    },
    Team {
      "abbreviation": "REES",
      "divisionStanding": 3,
      "firstName": "Feed",
      "lastName": "Zeeeke ",
      "leagueId": 336358,
      "leagueStanding": 4,
      "losses": 6,
      "pointsAgainst": 1290.12,
      "pointsFor": 1297.26,
      "seasonId": 2017,
      "teamId": 4,
      "ties": 0,
      "waiverRank": 7,
      "wins": 7,
    },
    Team {
      "abbreviation": "fuck",
      "divisionStanding": 2,
      "firstName": "fuck fantasy ",
      "lastName": "football",
      "leagueId": 336358,
      "leagueStanding": 5,
      "losses": 7,
      "pointsAgainst": 1278.42,
      "pointsFor": 1233.26,
      "seasonId": 2017,
      "teamId": 5,
      "ties": 0,
      "waiverRank": 4,
      "wins": 6,
    },
    Team {
      "abbreviation": "BOIS",
      "divisionStanding": 3,
      "firstName": "Team",
      "lastName": "ASS",
      "leagueId": 336358,
      "leagueStanding": 6,
      "losses": 8,
      "pointsAgainst": 1240.04,
      "pointsFor": 1328.1,
      "seasonId": 2017,
      "teamId": 8,
      "ties": 0,
      "waiverRank": 1,
      "wins": 5,
    },
    Team {
      "abbreviation": "��",
      "divisionStanding": 2,
      "firstName": "2 Gurleys",
      "lastName": "One Cup",
      "leagueId": 336358,
      "leagueStanding": 3,
      "losses": 5,
      "pointsAgainst": 1186.94,
      "pointsFor": 1288.64,
      "seasonId": 2017,
      "teamId": 9,
      "ties": 0,
      "waiverRank": 2,
      "wins": 8,
    },
    Team {
      "abbreviation": "YOC",
      "divisionStanding": 1,
      "firstName": "FRICKIN",
      "lastName": "JOOOOOSH!",
      "leagueId": 336358,
      "leagueStanding": 2,
      "losses": 6,
      "pointsAgainst": 1250.94,
      "pointsFor": 1219.6,
      "seasonId": 2017,
      "teamId": 12,
      "ties": 0,
      "waiverRank": 3,
      "wins": 7,
    },
  ],
  "firstMatchupId": 1,
  "firstWeekId": 1,
  "lastMatchupId": 15,
  "lastWeekId": 17,
  "leagueId": 336358,
  "name": "The Ocho",
  "numPlayoffTeams": 4,
  "numRegularSeasonMatchups": 13,
  "numTeams": 8,
  "playoffMatchupLength": 2,
  "playoffSeedOrder": Array [
    Team {
      "abbreviation": "GOAT",
      "divisionStanding": 1,
      "firstName": "The Boston",
      "lastName": "TE Party",
      "leagueId": 336358,
      "leagueStanding": 1,
      "losses": 4,
      "pointsAgainst": 1131.4,
      "pointsFor": 1305.78,
      "seasonId": 2017,
      "teamId": 1,
      "ties": 0,
      "waiverRank": 6,
      "wins": 9,
    },
    Team {
      "abbreviation": "GRON",
      "divisionStanding": 4,
      "firstName": "Prestige",
      "lastName": "Worldwide",
      "leagueId": 336358,
      "leagueStanding": 8,
      "losses": 8,
      "pointsAgainst": 1232.94,
      "pointsFor": 1062.82,
      "seasonId": 2017,
      "teamId": 2,
      "ties": 0,
      "waiverRank": 5,
      "wins": 5,
    },
    Team {
      "abbreviation": "ARCH",
      "divisionStanding": 4,
      "firstName": "Archie",
      "lastName": "Cooper",
      "leagueId": 336358,
      "leagueStanding": 7,
      "losses": 8,
      "pointsAgainst": 1280,
      "pointsFor": 1155.34,
      "seasonId": 2017,
      "teamId": 3,
      "ties": 0,
      "waiverRank": 8,
      "wins": 5,
    },
    Team {
      "abbreviation": "REES",
      "divisionStanding": 3,
      "firstName": "Feed",
      "lastName": "Zeeeke ",
      "leagueId": 336358,
      "leagueStanding": 4,
      "losses": 6,
      "pointsAgainst": 1290.12,
      "pointsFor": 1297.26,
      "seasonId": 2017,
      "teamId": 4,
      "ties": 0,
      "waiverRank": 7,
      "wins": 7,
    },
    Team {
      "abbreviation": "fuck",
      "divisionStanding": 2,
      "firstName": "fuck fantasy ",
      "lastName": "football",
      "leagueId": 336358,
      "leagueStanding": 5,
      "losses": 7,
      "pointsAgainst": 1278.42,
      "pointsFor": 1233.26,
      "seasonId": 2017,
      "teamId": 5,
      "ties": 0,
      "waiverRank": 4,
      "wins": 6,
    },
    Team {
      "abbreviation": "BOIS",
      "divisionStanding": 3,
      "firstName": "Team",
      "lastName": "ASS",
      "leagueId": 336358,
      "leagueStanding": 6,
      "losses": 8,
      "pointsAgainst": 1240.04,
      "pointsFor": 1328.1,
      "seasonId": 2017,
      "teamId": 8,
      "ties": 0,
      "waiverRank": 1,
      "wins": 5,
    },
    Team {
      "abbreviation": "��",
      "divisionStanding": 2,
      "firstName": "2 Gurleys",
      "lastName": "One Cup",
      "leagueId": 336358,
      "leagueStanding": 3,
      "losses": 5,
      "pointsAgainst": 1186.94,
      "pointsFor": 1288.64,
      "seasonId": 2017,
      "teamId": 9,
      "ties": 0,
      "waiverRank": 2,
      "wins": 8,
    },
    Team {
      "abbreviation": "YOC",
      "divisionStanding": 1,
      "firstName": "FRICKIN",
      "lastName": "JOOOOOSH!",
      "leagueId": 336358,
      "leagueStanding": 2,
      "losses": 6,
      "pointsAgainst": 1250.94,
      "pointsFor": 1219.6,
      "seasonId": 2017,
      "teamId": 12,
      "ties": 0,
      "waiverRank": 3,
      "wins": 7,
    },
  ],
  "playoffTiebreaker": "Total points for",
  "regularSeasonMatchupLength": 1,
  "regularSeasonTiebreaker": "None",
  "scoringDecimalPlaces": 2,
  "seasonId": "2017",
  "teams": Array [
    Team {
      "abbreviation": "GOAT",
      "divisionStanding": 1,
      "firstName": "The Boston",
      "lastName": "TE Party",
      "leagueId": 336358,
      "leagueStanding": 1,
      "losses": 4,
      "pointsAgainst": 1131.4,
      "pointsFor": 1305.78,
      "seasonId": 2017,
      "teamId": 1,
      "ties": 0,
      "waiverRank": 6,
      "wins": 9,
    },
    Team {
      "abbreviation": "GRON",
      "divisionStanding": 4,
      "firstName": "Prestige",
      "lastName": "Worldwide",
      "leagueId": 336358,
      "leagueStanding": 8,
      "losses": 8,
      "pointsAgainst": 1232.94,
      "pointsFor": 1062.82,
      "seasonId": 2017,
      "teamId": 2,
      "ties": 0,
      "waiverRank": 5,
      "wins": 5,
    },
    Team {
      "abbreviation": "ARCH",
      "divisionStanding": 4,
      "firstName": "Archie",
      "lastName": "Cooper",
      "leagueId": 336358,
      "leagueStanding": 7,
      "losses": 8,
      "pointsAgainst": 1280,
      "pointsFor": 1155.34,
      "seasonId": 2017,
      "teamId": 3,
      "ties": 0,
      "waiverRank": 8,
      "wins": 5,
    },
    Team {
      "abbreviation": "REES",
      "divisionStanding": 3,
      "firstName": "Feed",
      "lastName": "Zeeeke ",
      "leagueId": 336358,
      "leagueStanding": 4,
      "losses": 6,
      "pointsAgainst": 1290.12,
      "pointsFor": 1297.26,
      "seasonId": 2017,
      "teamId": 4,
      "ties": 0,
      "waiverRank": 7,
      "wins": 7,
    },
    Team {
      "abbreviation": "fuck",
      "divisionStanding": 2,
      "firstName": "fuck fantasy ",
      "lastName": "football",
      "leagueId": 336358,
      "leagueStanding": 5,
      "losses": 7,
      "pointsAgainst": 1278.42,
      "pointsFor": 1233.26,
      "seasonId": 2017,
      "teamId": 5,
      "ties": 0,
      "waiverRank": 4,
      "wins": 6,
    },
    Team {
      "abbreviation": "BOIS",
      "divisionStanding": 3,
      "firstName": "Team",
      "lastName": "ASS",
      "leagueId": 336358,
      "leagueStanding": 6,
      "losses": 8,
      "pointsAgainst": 1240.04,
      "pointsFor": 1328.1,
      "seasonId": 2017,
      "teamId": 8,
      "ties": 0,
      "waiverRank": 1,
      "wins": 5,
    },
    Team {
      "abbreviation": "��",
      "divisionStanding": 2,
      "firstName": "2 Gurleys",
      "lastName": "One Cup",
      "leagueId": 336358,
      "leagueStanding": 3,
      "losses": 5,
      "pointsAgainst": 1186.94,
      "pointsFor": 1288.64,
      "seasonId": 2017,
      "teamId": 9,
      "ties": 0,
      "waiverRank": 2,
      "wins": 8,
    },
    Team {
      "abbreviation": "YOC",
      "divisionStanding": 1,
      "firstName": "FRICKIN",
      "lastName": "JOOOOOSH!",
      "leagueId": 336358,
      "leagueStanding": 2,
      "losses": 6,
      "pointsAgainst": 1250.94,
      "pointsFor": 1219.6,
      "seasonId": 2017,
      "teamId": 12,
      "ties": 0,
      "waiverRank": 3,
      "wins": 7,
    },
  ],
  "tradeRevisionHours": 48,
}
`;

exports[`League attribute population from server response parses data correctly 1`] = `
League {
  "allowsTrades": true,
  "divisions": Array [
    Object {
      "divisionId": 0,
      "name": "Division 1",
      "size": 4,
    },
    Object {
      "divisionId": 1,
      "name": "Division 2",
      "size": 4,
    },
  ],
  "draftOrder": Array [
    Team {
      "abbreviation": "��",
      "divisionStanding": 2,
      "firstName": "2 Gurleys",
      "lastName": "One Cup",
      "leagueId": 336358,
      "leagueStanding": 3,
      "losses": 5,
      "pointsAgainst": 1186.94,
      "pointsFor": 1288.64,
      "seasonId": 2017,
      "teamId": 9,
      "ties": 0,
      "waiverRank": 2,
      "wins": 8,
    },
    Team {
      "abbreviation": "BOIS",
      "divisionStanding": 3,
      "firstName": "Team",
      "lastName": "ASS",
      "leagueId": 336358,
      "leagueStanding": 6,
      "losses": 8,
      "pointsAgainst": 1240.04,
      "pointsFor": 1328.1,
      "seasonId": 2017,
      "teamId": 8,
      "ties": 0,
      "waiverRank": 1,
      "wins": 5,
    },
    Team {
      "abbreviation": "REES",
      "divisionStanding": 3,
      "firstName": "Feed",
      "lastName": "Zeeeke ",
      "leagueId": 336358,
      "leagueStanding": 4,
      "losses": 6,
      "pointsAgainst": 1290.12,
      "pointsFor": 1297.26,
      "seasonId": 2017,
      "teamId": 4,
      "ties": 0,
      "waiverRank": 7,
      "wins": 7,
    },
    Team {
      "abbreviation": "YOC",
      "divisionStanding": 1,
      "firstName": "FRICKIN",
      "lastName": "JOOOOOSH!",
      "leagueId": 336358,
      "leagueStanding": 2,
      "losses": 6,
      "pointsAgainst": 1250.94,
      "pointsFor": 1219.6,
      "seasonId": 2017,
      "teamId": 12,
      "ties": 0,
      "waiverRank": 3,
      "wins": 7,
    },
    Team {
      "abbreviation": "ARCH",
      "divisionStanding": 4,
      "firstName": "Archie",
      "lastName": "Cooper",
      "leagueId": 336358,
      "leagueStanding": 7,
      "losses": 8,
      "pointsAgainst": 1280,
      "pointsFor": 1155.34,
      "seasonId": 2017,
      "teamId": 3,
      "ties": 0,
      "waiverRank": 8,
      "wins": 5,
    },
    Team {
      "abbreviation": "GOAT",
      "divisionStanding": 1,
      "firstName": "The Boston",
      "lastName": "TE Party",
      "leagueId": 336358,
      "leagueStanding": 1,
      "losses": 4,
      "pointsAgainst": 1131.4,
      "pointsFor": 1305.78,
      "seasonId": 2017,
      "teamId": 1,
      "ties": 0,
      "waiverRank": 6,
      "wins": 9,
    },
    Team {
      "abbreviation": "fuck",
      "divisionStanding": 2,
      "firstName": "fuck fantasy ",
      "lastName": "football",
      "leagueId": 336358,
      "leagueStanding": 5,
      "losses": 7,
      "pointsAgainst": 1278.42,
      "pointsFor": 1233.26,
      "seasonId": 2017,
      "teamId": 5,
      "ties": 0,
      "waiverRank": 4,
      "wins": 6,
    },
    Team {
      "abbreviation": "GRON",
      "divisionStanding": 4,
      "firstName": "Prestige",
      "lastName": "Worldwide",
      "leagueId": 336358,
      "leagueStanding": 8,
      "losses": 8,
      "pointsAgainst": 1232.94,
      "pointsFor": 1062.82,
      "seasonId": 2017,
      "teamId": 2,
      "ties": 0,
      "waiverRank": 5,
      "wins": 5,
    },
  ],
  "finalRankings": Array [
    Team {
      "abbreviation": "GOAT",
      "divisionStanding": 1,
      "firstName": "The Boston",
      "lastName": "TE Party",
      "leagueId": 336358,
      "leagueStanding": 1,
      "losses": 4,
      "pointsAgainst": 1131.4,
      "pointsFor": 1305.78,
      "seasonId": 2017,
      "teamId": 1,
      "ties": 0,
      "waiverRank": 6,
      "wins": 9,
    },
    Team {
      "abbreviation": "��",
      "divisionStanding": 2,
      "firstName": "2 Gurleys",
      "lastName": "One Cup",
      "leagueId": 336358,
      "leagueStanding": 3,
      "losses": 5,
      "pointsAgainst": 1186.94,
      "pointsFor": 1288.64,
      "seasonId": 2017,
      "teamId": 9,
      "ties": 0,
      "waiverRank": 2,
      "wins": 8,
    },
    Team {
      "abbreviation": "REES",
      "divisionStanding": 3,
      "firstName": "Feed",
      "lastName": "Zeeeke ",
      "leagueId": 336358,
      "leagueStanding": 4,
      "losses": 6,
      "pointsAgainst": 1290.12,
      "pointsFor": 1297.26,
      "seasonId": 2017,
      "teamId": 4,
      "ties": 0,
      "waiverRank": 7,
      "wins": 7,
    },
    Team {
      "abbreviation": "YOC",
      "divisionStanding": 1,
      "firstName": "FRICKIN",
      "lastName": "JOOOOOSH!",
      "leagueId": 336358,
      "leagueStanding": 2,
      "losses": 6,
      "pointsAgainst": 1250.94,
      "pointsFor": 1219.6,
      "seasonId": 2017,
      "teamId": 12,
      "ties": 0,
      "waiverRank": 3,
      "wins": 7,
    },
    Team {
      "abbreviation": "BOIS",
      "divisionStanding": 3,
      "firstName": "Team",
      "lastName": "ASS",
      "leagueId": 336358,
      "leagueStanding": 6,
      "losses": 8,
      "pointsAgainst": 1240.04,
      "pointsFor": 1328.1,
      "seasonId": 2017,
      "teamId": 8,
      "ties": 0,
      "waiverRank": 1,
      "wins": 5,
    },
    Team {
      "abbreviation": "GRON",
      "divisionStanding": 4,
      "firstName": "Prestige",
      "lastName": "Worldwide",
      "leagueId": 336358,
      "leagueStanding": 8,
      "losses": 8,
      "pointsAgainst": 1232.94,
      "pointsFor": 1062.82,
      "seasonId": 2017,
      "teamId": 2,
      "ties": 0,
      "waiverRank": 5,
      "wins": 5,
    },
    Team {
      "abbreviation": "fuck",
      "divisionStanding": 2,
      "firstName": "fuck fantasy ",
      "lastName": "football",
      "leagueId": 336358,
      "leagueStanding": 5,
      "losses": 7,
      "pointsAgainst": 1278.42,
      "pointsFor": 1233.26,
      "seasonId": 2017,
      "teamId": 5,
      "ties": 0,
      "waiverRank": 4,
      "wins": 6,
    },
    Team {
      "abbreviation": "ARCH",
      "divisionStanding": 4,
      "firstName": "Archie",
      "lastName": "Cooper",
      "leagueId": 336358,
      "leagueStanding": 7,
      "losses": 8,
      "pointsAgainst": 1280,
      "pointsFor": 1155.34,
      "seasonId": 2017,
      "teamId": 3,
      "ties": 0,
      "waiverRank": 8,
      "wins": 5,
    },
  ],
  "firstMatchupId": 1,
  "firstWeekId": 1,
  "lastMatchupId": 15,
  "lastWeekId": 17,
  "leagueId": 336358,
  "name": "The Ocho",
  "numPlayoffTeams": 4,
  "numRegularSeasonMatchups": 13,
  "numTeams": 8,
  "playoffMatchupLength": 2,
  "playoffSeedOrder": Array [
    Team {
      "abbreviation": "GOAT",
      "divisionStanding": 1,
      "firstName": "The Boston",
      "lastName": "TE Party",
      "leagueId": 336358,
      "leagueStanding": 1,
      "losses": 4,
      "pointsAgainst": 1131.4,
      "pointsFor": 1305.78,
      "seasonId": 2017,
      "teamId": 1,
      "ties": 0,
      "waiverRank": 6,
      "wins": 9,
    },
    Team {
      "abbreviation": "YOC",
      "divisionStanding": 1,
      "firstName": "FRICKIN",
      "lastName": "JOOOOOSH!",
      "leagueId": 336358,
      "leagueStanding": 2,
      "losses": 6,
      "pointsAgainst": 1250.94,
      "pointsFor": 1219.6,
      "seasonId": 2017,
      "teamId": 12,
      "ties": 0,
      "waiverRank": 3,
      "wins": 7,
    },
    Team {
      "abbreviation": "��",
      "divisionStanding": 2,
      "firstName": "2 Gurleys",
      "lastName": "One Cup",
      "leagueId": 336358,
      "leagueStanding": 3,
      "losses": 5,
      "pointsAgainst": 1186.94,
      "pointsFor": 1288.64,
      "seasonId": 2017,
      "teamId": 9,
      "ties": 0,
      "waiverRank": 2,
      "wins": 8,
    },
    Team {
      "abbreviation": "REES",
      "divisionStanding": 3,
      "firstName": "Feed",
      "lastName": "Zeeeke ",
      "leagueId": 336358,
      "leagueStanding": 4,
      "losses": 6,
      "pointsAgainst": 1290.12,
      "pointsFor": 1297.26,
      "seasonId": 2017,
      "teamId": 4,
      "ties": 0,
      "waiverRank": 7,
      "wins": 7,
    },
    Team {
      "abbreviation": "fuck",
      "divisionStanding": 2,
      "firstName": "fuck fantasy ",
      "lastName": "football",
      "leagueId": 336358,
      "leagueStanding": 5,
      "losses": 7,
      "pointsAgainst": 1278.42,
      "pointsFor": 1233.26,
      "seasonId": 2017,
      "teamId": 5,
      "ties": 0,
      "waiverRank": 4,
      "wins": 6,
    },
    Team {
      "abbreviation": "BOIS",
      "divisionStanding": 3,
      "firstName": "Team",
      "lastName": "ASS",
      "leagueId": 336358,
      "leagueStanding": 6,
      "losses": 8,
      "pointsAgainst": 1240.04,
      "pointsFor": 1328.1,
      "seasonId": 2017,
      "teamId": 8,
      "ties": 0,
      "waiverRank": 1,
      "wins": 5,
    },
    Team {
      "abbreviation": "ARCH",
      "divisionStanding": 4,
      "firstName": "Archie",
      "lastName": "Cooper",
      "leagueId": 336358,
      "leagueStanding": 7,
      "losses": 8,
      "pointsAgainst": 1280,
      "pointsFor": 1155.34,
      "seasonId": 2017,
      "teamId": 3,
      "ties": 0,
      "waiverRank": 8,
      "wins": 5,
    },
    Team {
      "abbreviation": "GRON",
      "divisionStanding": 4,
      "firstName": "Prestige",
      "lastName": "Worldwide",
      "leagueId": 336358,
      "leagueStanding": 8,
      "losses": 8,
      "pointsAgainst": 1232.94,
      "pointsFor": 1062.82,
      "seasonId": 2017,
      "teamId": 2,
      "ties": 0,
      "waiverRank": 5,
      "wins": 5,
    },
  ],
  "playoffTiebreaker": "Total points for",
  "regularSeasonMatchupLength": 1,
  "regularSeasonTiebreaker": "None",
  "scoringDecimalPlaces": 2,
  "seasonId": 2017,
  "teams": Array [
    Team {
      "abbreviation": "GOAT",
      "divisionStanding": 1,
      "firstName": "The Boston",
      "lastName": "TE Party",
      "leagueId": 336358,
      "leagueStanding": 1,
      "losses": 4,
      "pointsAgainst": 1131.4,
      "pointsFor": 1305.78,
      "seasonId": 2017,
      "teamId": 1,
      "ties": 0,
      "waiverRank": 6,
      "wins": 9,
    },
    Team {
      "abbreviation": "GRON",
      "divisionStanding": 4,
      "firstName": "Prestige",
      "lastName": "Worldwide",
      "leagueId": 336358,
      "leagueStanding": 8,
      "losses": 8,
      "pointsAgainst": 1232.94,
      "pointsFor": 1062.82,
      "seasonId": 2017,
      "teamId": 2,
      "ties": 0,
      "waiverRank": 5,
      "wins": 5,
    },
    Team {
      "abbreviation": "ARCH",
      "divisionStanding": 4,
      "firstName": "Archie",
      "lastName": "Cooper",
      "leagueId": 336358,
      "leagueStanding": 7,
      "losses": 8,
      "pointsAgainst": 1280,
      "pointsFor": 1155.34,
      "seasonId": 2017,
      "teamId": 3,
      "ties": 0,
      "waiverRank": 8,
      "wins": 5,
    },
    Team {
      "abbreviation": "REES",
      "divisionStanding": 3,
      "firstName": "Feed",
      "lastName": "Zeeeke ",
      "leagueId": 336358,
      "leagueStanding": 4,
      "losses": 6,
      "pointsAgainst": 1290.12,
      "pointsFor": 1297.26,
      "seasonId": 2017,
      "teamId": 4,
      "ties": 0,
      "waiverRank": 7,
      "wins": 7,
    },
    Team {
      "abbreviation": "fuck",
      "divisionStanding": 2,
      "firstName": "fuck fantasy ",
      "lastName": "football",
      "leagueId": 336358,
      "leagueStanding": 5,
      "losses": 7,
      "pointsAgainst": 1278.42,
      "pointsFor": 1233.26,
      "seasonId": 2017,
      "teamId": 5,
      "ties": 0,
      "waiverRank": 4,
      "wins": 6,
    },
    Team {
      "abbreviation": "BOIS",
      "divisionStanding": 3,
      "firstName": "Team",
      "lastName": "ASS",
      "leagueId": 336358,
      "leagueStanding": 6,
      "losses": 8,
      "pointsAgainst": 1240.04,
      "pointsFor": 1328.1,
      "seasonId": 2017,
      "teamId": 8,
      "ties": 0,
      "waiverRank": 1,
      "wins": 5,
    },
    Team {
      "abbreviation": "��",
      "divisionStanding": 2,
      "firstName": "2 Gurleys",
      "lastName": "One Cup",
      "leagueId": 336358,
      "leagueStanding": 3,
      "losses": 5,
      "pointsAgainst": 1186.94,
      "pointsFor": 1288.64,
      "seasonId": 2017,
      "teamId": 9,
      "ties": 0,
      "waiverRank": 2,
      "wins": 8,
    },
    Team {
      "abbreviation": "YOC",
      "divisionStanding": 1,
      "firstName": "FRICKIN",
      "lastName": "JOOOOOSH!",
      "leagueId": 336358,
      "leagueStanding": 2,
      "losses": 6,
      "pointsAgainst": 1250.94,
      "pointsFor": 1219.6,
      "seasonId": 2017,
      "teamId": 12,
      "ties": 0,
      "waiverRank": 3,
      "wins": 7,
    },
  ],
  "tradeRevisionHours": 48,
}
`;
